### PR TITLE
Cleanup mono debugging code

### PR DESCRIPTION
**Pull request by winterheart**
## Pull Request Type

- [ ] GitHub Workflow changes
- [ ] Documentation or Wiki changes
- [x] Build and Dependency changes
- [ ] Runtime changes
  - [ ] Render changes
  - [ ] Audio changes
  - [ ] Input changes
  - [ ] Network changes
  - [ ] Other changes

### Description

Remove unused mono debug code.

### Checklist

- [x] I have tested my changes locally and verified that they work as intended.
- [x] I have documented any new or modified functionality.
- [x] I have reviewed the changes to ensure they do not introduce any unnecessary complexity or duplicate code.
- [x] I understand that by submitting this pull request, I am agreeing to license my contributions under the project's license.

### CODE REVIEW

File: ddebug/debug.cpp [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+#include <cstdlib>
+#include <cstring>
+
+#include "debug.h"
+
+void Debug_LogClose();
+
+FILE *Debug_logfile = nullptr;
+
+bool Debug_Logfile(const char *filename) {
```
Comment (Lgt2x) [inline ddebug/debug.cpp:30]: Should we merge this with our logging system set up at `misc/log.cpp` ?

Comment (Lgt2x) [reply ddebug/debug.cpp:30]: or did I misunderstand the purpose of this module

Comment (winterheart) [reply ddebug/debug.cpp:30]: Yes, it is, but our logging facility still not written, so I decided left it as is for a while.

Comment (Lgt2x) [reply ddebug/debug.cpp:30]: fair enough. How can I test this change?

Comment (winterheart) [reply ddebug/debug.cpp:30]: Log file is enabled by `-logfile` option. After starting logfile will be created in current directory under "d3.log" filename. Content will be identical to console output.

Comment (Lgt2x) [reply ddebug/debug.cpp:30]: file is created but empty on Linux, is that expected?

Comment (winterheart) [reply ddebug/debug.cpp:30]: -DLOGGER need to be enabled too as mprintf() is enabled only under  LOGGER define. This is works only in Debug build.

Comment (Lgt2x) [reply ddebug/debug.cpp:30]: yes my build is Debug and ENABLE_LOGGER is ON, investigating

Comment (Lgt2x) [reply ddebug/debug.cpp:30]: `Debug_ConsolePrintf` seems to call `Debug_LogWrite` only on Windows (winmono) and not on Linux (lnxmono)

Comment (Lgt2x) [reply ddebug/debug.cpp:30]: Confirming that it works when adding `  Debug_LogWrite(Mono_buffer);` in `Debug_ConsolePrintf` in `lnxmono.cpp`

Comment (winterheart) [reply ddebug/debug.cpp:30]: Seems this code was broken at first place. Added `Debug_LogWrite(Mono_buffer)` call in `Debug_ConsolePrintf` for Linux.

**Review by Lgt2x — APPROVED** [review]
Now working on Linux

**Pull request merged**